Tasks/AzureAppServiceManageV0/azureappservicemanage.ts

```diff
@@ -88,8 +88,9 @@ async function run() {
 
             tl.debug(`Resource Group: ${resourceGroupName}`);
 
-            let slotNameNotNeeded: boolean = (action == "Complete Swap" || action == "Swap Slots" || action == "Start Swap With Preview");
-            slotName = (!specifySlotFlag || slotNameNotNeeded) ? defaultslotname : slotName;
+            if(!(((action == "Start Azure App Service" || action == "Stop Azure App Service" || action == "Restart Azure App Service") && !!specifySlotFlag) || action == "Delete Slot" || action == "Cancel Swap")) {
+                slotName = defaultslotname;
```

Comment:
Split the conditions and make it more readable. Add comments also to avoid further regrassion.
In master branch, please plan to add L0 test case to catch such regrassions in future.